src/main/java/duke/task/Task.java

```diff
@@ -0,0 +1,62 @@
+package duke.task;
+
+/**
+ * Represents a generic task in the Duke application.
+ */
+public class Task {
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Constructs a Task object with the specified description.
+     *
+     * @param description The description of the task.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
+
+    /**
+     * Returns the status icon of the task.
+     *
+     * @return The status icon of the task.
+     */
+    public String getStatusIcon() {
+        return (isDone ? "X" : " "); // mark done task with X
+    }
+
+    /**
+     * Marks the task as done by setting the 'isDone' flag to true.
+     */
+    public void markAsDone() {
```

Comment:
Maybe the method name could be changed to something like setDone to adhere to Java coding standards, as it is a setter method for a Boolean variable?
```suggestion
    public void setDone() {
```